ql/src/java/org/apache/hadoop/hive/ql/ddl/process/show/compactions/ShowCompactionsOperation.java

```diff
@@ -68,6 +76,37 @@ public int execute() throws HiveException {
     return 0;
   }
 
+  private ShowCompactRequest getShowCompactioRequest(ShowCompactionsDesc desc) throws MetaException, SemanticException {
+    ShowCompactRequest request = new ShowCompactRequest();
+    if (isBlank(desc.getDbName() )&& isNotBlank(desc.getTbName())) {
+      request.setDbname(SessionState.get().getCurrentDatabase());
+    } else {
+      request.setDbname(desc.getDbName());
+    }
+    if (isNotBlank(desc.getTbName()))
+      request.setTablename(desc.getTbName());
+    if (isNotBlank(desc.getPoolName()))
+      request.setPoolName(desc.getPoolName());
+    if (isNotBlank(desc.getCompactionType()))
+      request.setType(TxnUtils.inputCompactionType2DBType(desc.getCompactionType()));
+    if (isNotBlank(desc.getCompactionStatus()))
+      request.setState(TxnUtils.compactorStateFromInput(desc.getCompactionStatus()));
+    if (isNotEmpty(desc.getPartSpec()))
+      request.setPartitionname(getPartitionName(desc.getPartSpec()));
+    return request;
+  }
+
+  private String getPartitionName(Map<String, String> partitionSpec) throws SemanticException {
```

Comment:
same method exists in AlterTableUpdateColumnStatistictAnalyzer, could we move it to utils class?